runtime/onert/core/include/util/Index.h

```diff
@@ -138,6 +138,8 @@ template <typename T, typename DummyTag> class Index
    */
   T value() const { return _index; }
 
+  bool operator<(const Index &I) const { return value() < I.value(); }
```

Comment:
It requires `T` has total ordering. It may be okay because:

1) until now, `T` is integral type like `uint32_t`.
2) It already have `operator++`.

Reply:
I also plan to use this on a subsequence commit for draft https://github.com/Samsung/ONE/pull/12621

https://github.com/Samsung/ONE/blob/cc4ecce5a40c15cfab7ea41f1559feae63c2b3f0/runtime/onert/core/src/ir/train/TrainableGraph.cc#L239